spring-cloud-zookeeper-config/src/main/java/org/springframework/cloud/zookeeper/config/ZookeeperConfigBootstrapConfiguration.java

```diff
@@ -42,4 +48,35 @@ public ZookeeperPropertySourceLocator zookeeperPropertySourceLocator(
 	public ZookeeperConfigProperties zookeeperConfigProperties() {
 		return new ZookeeperConfigProperties();
 	}
+
+	@Bean
+	public ApplicationListener<ContextRefreshedEvent> zookeeperTreeCachePropertySourceContextRefreshedApplicationListener() {
```

Comment:
Regardless of the approach (still wrapping my head around it), this doesn't seem like the right way to set the application context. If it had to be set (again not sure), I'd set it when the property source is created.